### PR TITLE
Added code to pull dashboard version automatically for E2E tests

### PR DESCRIPTION
**Pull request by amulyavarote**
# Description

Added code to get the latest Dapr dashboard version automatically for E2E tests

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #832 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
@amulyavarote Please modify the E2E tests also the use a specified dashboard version.

**Comment by mukundansundar** [comment]
@amulyavarote Take a look at https://github.com/dapr/quickstarts/blob/master/.github/workflows/validate.yaml#L90 and add similar functionality in the Yaml files for the E2E in CLI to pull the latest RC/Stable release of Runtime and dashboard, and make use of the passed in versions from Yaml files in the Tests ....

**Comment by mukundansundar** [comment]
@amulyavarote Still the version 1.5.0 is what is being used please remove the overrides that are there for versions in the standalone_test.go and Kubernetes_test.go in tests/e2e

File: .github/workflows/kind_e2e.yaml [hunk 1]
```diff
@@ -121,7 +121,18 @@ jobs:
     - name: Free up some diskspace
       run: |
         docker image prune -a -f
-
+    - name: Determine latest Dapr Runtime version including Pre-releases
```
Comment (artursouza) [inline .github/workflows/kind_e2e.yaml:124]: @mukundansundar Would this make builds not reproducible for release branches going forward? I remember pinning the runtime version in our Java SDK IT tests to make sure old releases continue to build the same way.

File: cmd/init.go [hunk 2]
```diff
@@ -100,12 +100,18 @@ func init() {
 	if runtimeVersionEnv != "" {
 		defaultRuntimeVersion = runtimeVersionEnv
 	}
+	defaultDashboardVersion := "latest"
```
Comment (artursouza) [inline cmd/init.go:103]: LGTM

File: .github/workflows/kind_e2e.yaml (hunk 1 above)
Comment (artursouza) [reply .github/workflows/kind_e2e.yaml:124]: I would keep the other changes but pin the runtime and dashboard versions here instead.

Comment (artursouza) [reply .github/workflows/kind_e2e.yaml:124]: Another option is to combine both where if in master branch use the latest but if from a release branch, use a pinned version.

Comment (mukundansundar) [reply .github/workflows/kind_e2e.yaml:124]: I agree @artursouza. Pinning the versions in the release branch and using the latest versions in the master branch and associated PRs makes perfect sense. 
@amulyavarote can you make changes such that PRs targeting release branch and the release branch itself have pinned versions whereas the master branch and PRs targeting master branch have the latest versions?

**Pull request merged**